staging/src/k8s.io/apiextensions-apiserver/pkg/controller/finalizer/crd_finalizer.go

```diff
@@ -193,7 +193,6 @@ func (c *CRDFinalizer) deleteInstances(crd *apiextensionsv1.CustomResourceDefini
 		}, err
 	}
 
-	ctx := genericapirequest.NewContext()
```

Comment:
`genericapirequest.NewContext()` == `context.TODO` - definitely worth replacing!

We should deprecate and remove `genericapirequest.NewContext` and `genericapirequest.NewDefaultContext`. The latter has to be replaced with something which accepts a context from its caller.